### DISCUSSION TITLE
Software Motion Detection

### DISCUSSION THREAD

**Discussion by ali86x** [arendst/Tasmota#20673] [Q&A] [answered]
Is it possible to add Motion Detection to a custom Firmware?
Maybe send a picture on Motion or a Mqtt Event?

**Comment by sfromis** [comment]
Such questions depends on details of what you mean. Adding a simple PIR is easy to do, but lots of technologies for motion/people detection exists.

**Comment by ali86x** [comment]
Okay. 
Adding a PIR sensor is a very easy way. That is what I did actually.
But maybe there is a solution in Tasmota for analysing picture frame by frame and then starting specific actions.
Maybe you know the Basic Arduino Templates for Esp32 Cam. There is also motion Detection integrated.
I would Like Something similar, but based in Tasmota

**Reply by fknaopen** [reply, marked as answer]
"The Tasmota webcam build enables a lot of camera features."
-->   https://github.com/btsimonh/TasmotaBerryPLay/blob/master/tasdocs/esp32cam.md
